File: 217-ContainsDuplicate/217-ContainsDuplicate.java
// Last updated: 11/2/2025, 8:23:35 PM
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> set = new HashSet<Integer>();

        for(int i:nums){
            if(!set.add(i)){
                return true;
            }
        }
        return false;
    }
}

File: 217-ContainsDuplicate/217-ContainsDuplicate.py
# Last updated: 11/2/2025, 8:23:34 PM
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        
        hashset = set()

        for n in nums:

            if n in hashset:
                return True 
            
            hashset.add(n)

        return False 
        
